otter/supervisor.py

```diff
@@ -683,3 +685,23 @@ def scrub_otter_metadata(_):
 
     d.addCallback(remove_server_from_state)
     return d
+
+
+@attributes(['scaling_group', 'server_id'])
+class EvictServerFromScalingGroup(object):
+    """
+    An Effect intent which indicates that a server should be evicted from a
+    particular group.
+    """
+
+
+@deferred_performer
+def perform_evict_server(log, transaction_id, dispatcher, intent):
```

Comment:
Not sure if its possible, but will be nice to receive `supervisor` as argument to the performer.